### PR TITLE
feat: Refactor `HttpErrorHandler` to make it consistent with Handlers implementation

### LINKED ISSUE #7280: Refactor `HttpErrorHandler` to make it consistent with Handlers implementation
`HttpErrorHandler` should implement Handler and share common interface and logic for handler resolution.

### PR DESCRIPTION
**Pull request by dunqan**
HttpErrorHandlernow implements Handler and shares common interface and logic for handler resolution.

Closes #7280

### CODE REVIEW

File: projects/core/src/global-message/http-interceptors/handlers/unknown-error/unknown-error.handler.ts [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
+  }
+
   handleError() {
     if (isDevMode()) {
       console.warn(`Unknown http response error: ${this.responseStatus}`);
     }
   }
+
+  getPriority() {
+    return Priority.FALLBACK;
```
Comment (tobi-or-not-tobi) [inline projects/core/src/global-message/http-interceptors/handlers/unknown-error/unknown-error.handler.ts:33]: can we add some docs to explain why this handler has a lower priority?

File: projects/core/src/global-message/http-interceptors/handlers/http-error.handler.ts [hunk 2]
```diff
@@ -24,4 +25,10 @@ export abstract class HttpErrorHandler {
     request: HttpRequest<any>,
     errorResponse: HttpErrorResponse
   ): void;
+
+  hasMatch(errorResponse: HttpErrorResponse): boolean {
```
Comment (tobi-or-not-tobi) [inline projects/core/src/global-message/http-interceptors/handlers/http-error.handler.ts:33]:
```suggestion
/**
 * Error handlers are matched by the error `responseStatus` (i.e. 404). On top of the matching status 
 * a priority can be added to distinguish multiple handles for the same response status. 
 */
  hasMatch(errorResponse: HttpErrorResponse): boolean {
```

File: projects/storefrontlib/src/cms-structure/page/component/services/component-handler.service.ts [hunk 3]
```diff
@@ -33,17 +33,9 @@ export class ComponentHandlerService {
...
-    );
-    if (matchedHandlers.length > 1) {
-      matchedHandlers.sort(
-        (a, b) =>
-          (a.getPriority ? a.getPriority() : 0) -
-          (b.getPriority ? b.getPriority() : 0)
-      );
-    }
-    if (isDevMode() && matchedHandlers.length === 0) {
+    const handler = resolveHandler(this.handlers, [componentMapping]);
```
Comment (tobi-or-not-tobi) [inline projects/storefrontlib/src/cms-structure/page/component/services/component-handler.service.ts:36]: I guess we sneaked this in ;)

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/src/cms-structure/page/component/services/component-handler.service.ts (hunk 3 above)
Comment (dunqan) [reply projects/storefrontlib/src/cms-structure/page/component/services/component-handler.service.ts:36]: yup, didn't want to create yet another PR just for this,

File: projects/core/src/global-message/http-interceptors/handlers/unknown-error/unknown-error.handler.ts (hunk 1 above)
Comment (dunqan) [reply projects/core/src/global-message/http-interceptors/handlers/unknown-error/unknown-error.handler.ts:33]: 👍

**Review by tobi-or-not-tobi — DISMISSED** [review]

**Review by marlass — APPROVED** [review]

**Pull request merged**